tasks/sysctl.yml

```diff
@@ -14,7 +14,7 @@
 - name: Set sysctl net.ipv6.conf.accept_ra_rtr_pref
   become: true
   ansible.posix.sysctl:
-    name: net.ipv6.conf.{{ ansible_default_ipv4.interface }}.accept_ra_rtr_pref
+    name: net.ipv6.conf.{{ ansible_default_ipv4.interface | regex_replace('\.', '/') }}.accept_ra_rtr_pref
```

Comment:
```console
Syntax Error while loading YAML.
  found unknown escape character
```

Reply:
I already test with `| regex_replace('\\.', '/')` it not working.
so that I use `| regex_replace('\.', '/')` instead.
Btw, My code already runs in my production without problem.

Could you share how to test or where an error came from? I will diagnose it again.

Reply:
```sh
$ ansible-playbook -i 'localhost,' -c local regsys.yml 
ERROR! We were unable to read either as JSON nor YAML, these are the errors we got from each:
JSON: Expecting value: line 1 column 1 (char 0)

Syntax Error while loading YAML.
  found unknown escape character

The error appears to be in '/home/tsj/regsys.yml': line 15, column 64, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
                                                               ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes. Always quote template expression brackets when they
start a value. For instance:

    with_items:
      - {{ foo }}

Should be written as:

    with_items:
      - "{{ foo }}"
$ cat regsys.yml 
---
- hosts: localhost
  gather_facts: true
  tasks:
    - name: A
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address }}"
          
    - name: B
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\\.', '/') }}"

    - name: C
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
$ vi regsys.yml 
$ cat regsys.yml 
---
- hosts: localhost
  gather_facts: true
  tasks:
    - name: A
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address }}"
          
    - name: B
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\\.', '/') }}"

#    - name: C
#      ansible.builtin.debug:
#        msg: "{{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
$ ansible-playbook -i 'localhost,' -c local regsys.yml 

PLAY [localhost] ************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************
ok: [localhost]

TASK [A] *********************************************************************************************************
ok: [localhost] => {
    "msg": "192.168.0.95"
}

TASK [B] *********************************************************************************************************
ok: [localhost] => {
    "msg": "192/168/0/95"
}

PLAY RECAP ***************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
````

Reply:
The error will occur in case we have quoted text by double quote. 
For example
```
# No error
    - name: C
      ansible.builtin.debug:
        msg: test {{ ansible_default_ipv4.address | regex_replace('\.', '/') }}
# Error
    - name: C
      ansible.builtin.debug:
        msg: "test {{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
```
According to this yaml spec https://yaml.org/spec/1.2.2/#731-double-quoted-style
Double-quote will read escape string but single quote does not.

I will add some tests and quote sysctl name with double quote.